Workflow API docs (#2915)
dapr/docs: daprdocs/content/en/developing-applications/building-blocks/workflow/workflow-overview.md

---
type: docs
title: Workflow overview
linkTitle: Overview
weight: 1000
description: "Overview of the workflow building block"
---

Workflows are quickly becoming a core element to cloud-native applications. A workflow is custom application logic that:

- Defines a business process or data flow that spans multiple microservices
- Drives the execution of these business processes to completion in a reliable way
- Consists of a set of tasks and/or state transitions, which can be either statically defined or dynamic
- May be authored using declarative markup or general purpose programming languages

> 

Reviewer: Reworked this section a bit.

---
type: docs
title: Workflow overview
linkTitle: Overview
weight: 1000
description: "Overview of the workflow building block"
---

> With Dapr workflows, you can automate and orchestrate tasks within your application. The workflow feature provides a developer-friendly programming model for authoring workflows as code in a way that abstracts away the complexities of messaging, state management, and failure handling.

Reviewer 2: ```suggestion
With workflows, you can orchestrate tasks or business logic within your application, making them reliable and restartable. Workflows provide a programming model for authoring code that abstracts away the complexities of messaging, state management, and failure handling.
```

> The workflow feature is comprised of the Workflow API and the workflow components that allow for authoring workflows naitively in Dapr or through external 3rd party workflow applications.

Reviewer 2: ```suggestion
```

> <img src="/images/concepts-building-blocks.png" width=250>

Reviewer 2: ```suggestion

```

> ## Workflow API

Reviewer 2: ```suggestion
## Workflow building block
```

> 

Reviewer 2: The workflow building block:
- Provides a workflow API for running workflows 
- An built-in workflow runtime to write Dapr workflows (of type `workflow.dapr`)
- Integrates with various workflow runtimes as components, for example Temporal workflows.

*To Do:*
- *Add diagram of how workflow API works?*

> - *Add explaination of how the API works*

Reviewer 3: <img src="/dapr/docs/daprdocs/static/images/workflows-diagram.png" width=1000>
Create an application and run it with the dapr sidecar. After running the application, HTTP/gRPC calls can be made to dapr to run specific tasks/workflows that reside in the application.

Each individual workflow can be started or terminated through a POST request. You can also get information on the workflow (even if it has been terminated or finished naturally!) through a GET request. This GET request will send back information such as the instance ID of the workflow as well as the time that the run started, and even the current running status whether that be “Running”, “Terminated” or “Completed”.

- *[Reference](https://docs.dapr.io/developing-applications/building-blocks/service-invocation/service-invocation-overview/)*

---

## Features

> *To Do*

Reviewer 2: ```suggestion
Start, Terminate and X any workflow
```

---
type: docs
title: Workflow overview
linkTitle: Overview
weight: 1000
description: "Overview of the workflow building block"
---

With Dapr workflows, you can automate and orchestrate tasks within your application. The workflow feature provides a developer-friendly programming model for authoring workflows as code in a way that abstracts away the complexities of messaging, state management, and failure handling.

The workflow feature is comprised of the Workflow API and the workflow components that allow for authoring workflows naitively in Dapr or through external 3rd party workflow applications.

<img src="/images/concepts-building-blocks.png" width=250>

## Workflow API

*To Do:*
- *Add diagram of how workflow API works?*
- *Add explaination of how the API works*
- *[Reference](https://docs.dapr.io/developing-applications/building-blocks/service-invocation/service-invocation-overview/)*

---

## Features

The workflow API brings several core features executed by the Dapr sidecar:

### Combine Dapr APIs
- Workflows allows users to combine Dapr APIs together in scheduled tasks

> - Workflows compatible with pubsub, state store, bindings 

Reviewer 3: "Workflows are compatible"?

> - Can send/respond to external signals, including pub/sub events and bindings (input or output

Reviewer 3: Nit: Close the parentheses at the end of this. Also, start the sentence with "Workflows" to be consistent with the two points above.
